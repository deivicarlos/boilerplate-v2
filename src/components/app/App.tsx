import { CSSProperties } from 'react';
import logo from './logo.svg';
import Counter from '../counter/Counter';

const appLink: CSSProperties = {
  color: `rgb(112, 76, 182)`,
};

const App = () => {
  return (
    <div className="text-center">
      <header className="min-h-screen flex flex-col items-center justify-center text-3xl">
        <img
          src={logo}
          className="animate-[bounce_infinite_3s_ease-in-out] h-52"
          alt="logo"
        />
        <Counter />
        <p className="mt-4">
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <span>
          <a
            style={appLink}
            href="https://reactjs.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            React
          </a>
          <span>, </span>
          <a
            style={appLink}
            href="https://redux.js.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Redux
          </a>
          <span>, </span>
          <a
            style={appLink}
            href="https://redux-toolkit.js.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Redux Toolkit
          </a>
          ,<span> and </span>
          <a
            style={appLink}
            href="https://react-redux.js.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            React Redux
          </a>
        </span>
      </header>
    </div>
  );
};

export default App;
