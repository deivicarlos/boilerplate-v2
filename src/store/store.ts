import {
  Action,
  combineReducers,
  configureStore,
  ThunkAction,
} from '@reduxjs/toolkit';
import counterReducer from './counter/reducer';

export const appReducer = combineReducers({
  counter: counterReducer,
});

export const storeX = configureStore({
  reducer: {
    counter: counterReducer,
  },
});

export const store = configureStore({
  reducer: appReducer,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
