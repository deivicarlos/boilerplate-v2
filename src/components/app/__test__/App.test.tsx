import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../store/store';
import App from '../App';

describe('App', () => {
  let asFragment: any;

  beforeEach(() => {
    const component = render(
      <Provider store={store}>
        <App />
      </Provider>
    );

    asFragment = component.asFragment;
  });

  it('renders correctly', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
