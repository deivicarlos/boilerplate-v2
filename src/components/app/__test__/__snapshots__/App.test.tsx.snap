// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="text-center"
  >
    <header
      class="min-h-screen flex flex-col items-center justify-center text-3xl"
    >
      <img
        alt="logo"
        class="animate-[bounce_infinite_3s_ease-in-out] h-52"
        src="logo.svg"
      />
      <div>
        <div
          class="row"
        >
          <button
            aria-label="Decrement value"
            class="button"
            type="button"
          >
            -
          </button>
          <span
            class="value"
          >
            0
          </span>
          <button
            aria-label="Increment value"
            class="button"
            type="button"
          >
            +
          </button>
        </div>
      </div>
      <p
        class="mt-4"
      >
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <span>
        <a
          href="https://reactjs.org/"
          rel="noopener noreferrer"
          style="color: rgb(112, 76, 182);"
          target="_blank"
        >
          React
        </a>
        <span>
          , 
        </span>
        <a
          href="https://redux.js.org/"
          rel="noopener noreferrer"
          style="color: rgb(112, 76, 182);"
          target="_blank"
        >
          Redux
        </a>
        <span>
          , 
        </span>
        <a
          href="https://redux-toolkit.js.org/"
          rel="noopener noreferrer"
          style="color: rgb(112, 76, 182);"
          target="_blank"
        >
          Redux Toolkit
        </a>
        ,
        <span>
           and 
        </span>
        <a
          href="https://react-redux.js.org/"
          rel="noopener noreferrer"
          style="color: rgb(112, 76, 182);"
          target="_blank"
        >
          React Redux
        </a>
      </span>
    </header>
  </div>
</DocumentFragment>
`;
